### PR TITLE
Send User.list response in private

### PR DESCRIPTION
**Pull request by jacksonmj**
Controlled by new config value plugins.User.listInPrivate.

This avoids mass-highlights of people who use their IRC nick as their bot username.

### CODE REVIEW

File: plugins/User/config.py [hunk 1]
```diff
@@ -43,8 +43,9 @@ def configure(advanced):
 
 
 User = conf.registerPlugin('User')
-# This is where your configuration variables (if any) should go.  For example:
-# conf.registerGlobalValue(User, 'someConfigVariableName',
-#     registry.Boolean(False, """Help for someConfigVariableName."""))
+conf.registerGlobalValue(User, 'listInPrivate',
```
Comment (ProgVal) [inline plugins/User/config.py]: Can you make it channel-specific?

File: plugins/User/config.py [hunk 2]
```diff
@@ -43,8 +43,9 @@ def configure(advanced):
...
+    registry.Boolean(True, _("""Determines whether the output of 'user list'
```
Comment (ProgVal) [inline plugins/User/config.py]: I would prefer the default to be False, so it follows the generic behavior by default

File: plugins/User/config.py [hunk 3]
```diff
@@ -43,8 +43,9 @@ def configure(advanced):
...
+    will be sent in private. This prevents mass-highlights of people who use
+    their nick as their bot username.""")))
```
Comment (ProgVal) [inline plugins/User/config.py]: You could add “This has no effect if supybot.reply.inPrivate is true”

File: plugins/User/config.py (hunk 2 above)
Comment (GLolol) [inline plugins/User/config.py]: #1042 is similar and already defaults to True.

Comment (ProgVal) [inline plugins/User/config.py]: good point

File: plugins/User/plugin.py [hunk 4]
```diff
@@ -88,7 +88,8 @@ def p(u):
                 users.append(u.name)
         if users:
             utils.sortBy(str.lower, users)
-            irc.reply(format('%L', users))
+            private = self.registryValue("listInPrivate")
```
Comment (ProgVal) [inline plugins/User/plugin.py]: `self.registryValue("listInPrivate", channel)` to make it channel-specific

File: plugins/User/config.py (hunk 3 above)
Comment (jacksonmj) [inline plugins/User/config.py]: The config value appears to override supybot.reply.inPrivate. 
https://github.com/ProgVal/Limnoria/blob/testing/src/callbacks.py#L174-L175 - supybot.reply.inPrivate only takes effect if the private argument is None, and is ignored if private is True or False.

**Pull request merged**